src/buildx/install.ts

```diff
@@ -32,28 +33,47 @@ import {Docker} from '../docker/docker';
 import {Git} from '../git';
 import {Util} from '../util';
 
-import {GitHubRelease} from '../types/github';
+import {GitHubRelease, OctokitRelease} from '../types/github';
 
 export interface InstallOpts {
   standalone?: boolean;
+  gitAuthToken?: string;
+  octokit?: InstanceType<typeof Octokit>;
 }
 
 export class Install {
   private readonly _standalone: boolean | undefined;
+  private readonly _gitAuthToken: string | undefined;
+  private readonly _octokit: InstanceType<typeof Octokit> | undefined;
 
   constructor(opts?: InstallOpts) {
     this._standalone = opts?.standalone;
+    this._gitAuthToken = opts?.gitAuthToken;
+    this._octokit = opts?.octokit;
   }
 
   /*
    * Download buildx binary from GitHub release
-   * @param version semver version or latest
+   * @param version semver version or latest.
+   * @param owner Repo owner, defaults to 'docker'.
+   * @param repo Repo name, defaults to 'buildx'.
    * @returns path to the buildx binary
    */
-  public async download(version: string): Promise<string> {
-    const release: GitHubRelease = await Install.getRelease(version);
-    core.debug(`Install.download release tag name: ${release.tag_name}`);
+  public async download(version: string, owner?: string, repo?: string): Promise<string> {
```

Comment:
We don't need to change the signature imo. As said in previous PR we can just set `version` to a tagged release URL like `https://github.com/docker/buildx/releases/tag/v0.11.1`.

Reply:
done! i changed this to accept a url of that form